File: Example/MachInjectorExample/TestFramework/Initialize.m
//
//  Initialize.m
//  TestFramework
//
//  Created by JH on 2026/1/16.
//

#include "Initialize.h"

extern void swift_initializeTestFramework(void);

static void initializeTestFramework(void) {
    swift_initializeTestFramework();
}

File: Example/MachInjectorExample/TestFramework/Initialize.swift
import Foundation
import os.log

private let logger = Logger(subsystem: "com.machinjector.testframework", category: "Initialization")

@_cdecl("swift_initializeTestFramework")
func initialize() {
    logger.info("TestFramework initialized in process \(ProcessInfo.processInfo.processName, privacy: .public) (PID: \(ProcessInfo.processInfo.processIdentifier, privacy: .public))")
}
